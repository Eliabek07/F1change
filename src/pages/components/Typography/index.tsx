import React from 'react';
import {Text, TextStyle} from 'react-native';

import {returnTypeFont} from '../../../utils/font_type';
import {returnTypeColors} from '../../../utils/color_type';

interface Props {
  variant?: keyof typeof returnTypeFont;
  color?: keyof typeof returnTypeColors;
  children?: any;
  style?: TextStyle;
}
function Typography({
  color = 'neutro_700',
  variant = '14_regular',
  children = null,
  style = {},
}: Props) {
  const returnStylingFont = () => {
    return returnTypeFont[variant];
  };
  return (
    <Text
      style={[
        returnStylingFont(),
        {color: returnTypeColors[color] as any},
        style,
      ]}>
      {children}
    </Text>
  );
}

export default Typography;
