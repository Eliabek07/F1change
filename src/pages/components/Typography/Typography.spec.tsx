import React from 'react';
import {render} from '@testing-library/react-native';
import Typography from '.';

describe('Testing Typography component', () => {
  it('Shoud render Typography', () => {
    const {toJSON, getByText} = render(
      <Typography variant="14_bold" color="alert_error_200">
        Testando Typography
      </Typography>,
    );
    expect(getByText(/Testando Typography/i)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
  it('Shoud render with empty values', () => {
    const {toJSON} = render(<Typography />);
    expect(toJSON()).toMatchSnapshot();
  });
});
