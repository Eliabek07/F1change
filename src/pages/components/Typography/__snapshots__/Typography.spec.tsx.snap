// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Typography component Shoud render Typography 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Inter-Bold",
        "fontSize": 14,
        "lineHeight": 20,
      },
      {
        "color": "#9AF4C3",
      },
      {},
    ]
  }
>
  Testando Typography
</Text>
`;

exports[`Testing Typography component Shoud render with empty values 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Inter-Regular",
        "fontSize": 14,
        "lineHeight": 20,
      },
      {
        "color": "#44403C",
      },
      {},
    ]
  }
/>
`;
